exchanges/kucoin/kucoin_futures.go

```diff
@@ -186,165 +164,123 @@ func (ku *Kucoin) GetFuturesInterestRate(ctx context.Context, symbol string, sta
 	}
 	if maxCount != 0 {
 		params.Set("maxCount", strconv.FormatInt(maxCount, 10))
 	}
-	return resp.Data.List, ku.SendHTTPRequest(ctx, exchange.RestFutures, common.EncodeURLValues(kucoinFuturesInterestRate, params), &resp)
-}
-
-// GetFuturesIndexList retrives futures index information for a symbol
-func (ku *Kucoin) GetFuturesIndexList(ctx context.Context, symbol string, startAt, endAt time.Time, reverse, forward bool, offset, maxCount int64) ([]FuturesIndex, error) {
 	resp := struct {
-		Data struct {
-			List    []FuturesIndex `json:"dataList"`
-			HasMore bool           `json:"hasMore"`
-		} `json:"data"`
-		Error
+		List    []FuturesInterestRate `json:"dataList"`
+		HasMore bool                  `json:"hasMore"`
 	}{}
+	return resp.List, ku.SendHTTPRequest(ctx, exchange.RestFutures, defaultFuturesEPL, common.EncodeURLValues(kucoinFuturesInterestRate, params), &resp)
+}
 
-	params := url.Values{}
+// GetFuturesIndexList retrieves futures index information for a symbol
+func (ku *Kucoin) GetFuturesIndexList(ctx context.Context, symbol string, startAt, endAt time.Time, reverse, forward bool, offset, maxCount int64) ([]FuturesIndex, error) {
 	if symbol == "" {
-		return resp.Data.List, errors.New("symbol can't be empty")
+		return nil, errors.New("symbol can't be empty")
 	}
+	params := url.Values{}
 	params.Set("symbol", symbol)
-
 	if !startAt.IsZero() {
 		params.Set("startAt", strconv.FormatInt(startAt.UnixMilli(), 10))
 	}
 	if !endAt.IsZero() {
 		params.Set("endAt", strconv.FormatInt(endAt.UnixMilli(), 10))
 	}
-	if reverse {
-		params.Set("reverse", "true")
-	} else {
-		params.Set("reverse", "false")
-	}
-	if forward {
-		params.Set("forward", "true")
-	} else {
-		params.Set("forward", "false")
-	}
+	params.Set("reverse", strconv.FormatBool(reverse))
+	params.Set("forward", strconv.FormatBool(forward))
 	if offset != 0 {
 		params.Set("offset", strconv.FormatInt(offset, 10))
 	}
 	if maxCount != 0 {
 		params.Set("maxCount", strconv.FormatInt(maxCount, 10))
 	}
-	return resp.Data.List, ku.SendHTTPRequest(ctx, exchange.RestFutures, common.EncodeURLValues(kucoinFuturesIndex, params), &resp)
+	resp := struct {
+		List    []FuturesIndex `json:"dataList"`
+		HasMore bool           `json:"hasMore"`
+	}{}
+	return resp.List, ku.SendHTTPRequest(ctx, exchange.RestFutures, defaultFuturesEPL, common.EncodeURLValues(kucoinFuturesIndex, params), &resp)
 }
 
 // GetFuturesCurrentMarkPrice get current mark price
 func (ku *Kucoin) GetFuturesCurrentMarkPrice(ctx context.Context, symbol string) (*FuturesMarkPrice, error) {
-	resp := struct {
-		Data FuturesMarkPrice `json:"data"`
-		Error
-	}{}
-
 	if symbol == "" {
 		return nil, errors.New("symbol can't be empty")
 	}
-	return &resp.Data, ku.SendHTTPRequest(ctx, exchange.RestFutures, fmt.Sprintf(kucoinFuturesMarkPrice, symbol), &resp)
+	var resp *FuturesMarkPrice
+	return resp, ku.SendHTTPRequest(ctx, exchange.RestFutures, defaultFuturesEPL, fmt.Sprintf(kucoinFuturesMarkPrice, symbol), &resp)
```

Comment:
concatenation may affect the reusability of the variable since it is going to be like this "/v1/mark-price/"+symbol+"/current" and we can't declare two variables for the right and left 😀 and reuse it.